### PR TITLE
Season summary da

### PR DESCRIPTION
**Pull request by d-atkins**
- Last ditch effort
- Add seasonal summary
- Add winningest/worst coach
- Spec harness runs in ~30 seconds, missing 8 methods

### CODE REVIEW

File: lib/season_stats.rb [hunk 1]
```diff
@@ -1,22 +1,37 @@
...
 
 class SeasonStats
   include Createable
 
-  def initialize(game_collection, game_teams_collection)
-    @game_collection = game_collection
-    @game_teams_collection = game_teams_collection
+  def initialize(game_stats, gtc, team_collection)
+    @game_stats = game_stats
+    @gtc = gtc
```
Comment (mronauli) [inline lib/season_stats.rb:8]: Nice. More readable imo

**Review by mronauli — COMMENTED** [review]
I like what you did!

File: lib/season_stats.rb [hunk 2]
```diff
@@ -9,12 +9,15 @@ def initialize(game_collection, game_teams_collection)
...
+
   def results_by_opponents(team_id)
     @game_teams_collection.game_teams_by_id[team_id].reduce({}) do |acc, game_team|
-      corresponding_game = @game_collection.games.find {|game| game.game_id == game_team.game_id }
-      opponent = corresponding_game.opponent_id(team_id)
-      acc[opponent] << game_team.result if acc[opponent]
-      acc[opponent] = [game_team.result] if acc[opponent].nil?
+      opponent_id = corresponding_game(game_team.game_id).opponent_id(team_id)
+      acc[opponent_id] << game_team.result if acc[opponent_id]
+      acc[opponent_id] = [game_team.result] if acc[opponent_id].nil?
```
Comment (Yetidancer) [inline lib/season_stats.rb]: Split out as helper method--makes sense

File: lib/season_stats.rb [hunk 3]
```diff
@@ -37,6 +40,27 @@ def make_season_game_array(season)
...
+  def format_seasonal_summary(split_hash)
+    reg_games = split_hash[:regular_season]
+    post_games = split_hash[:postseason]
+    split_hash[:regular_season] = {win_percentage: reg_games,
+      total_goals_scored: reg_games, total_goals_against: reg_games,
+      average_goals_scored: reg_games, average_goals_against: reg_games}
+    split_hash[:postseason] = {win_percentage: post_games,
+      total_goals_scored: post_games, total_goals_against: post_games,
+      average_goals_scored: post_games, average_goals_against: post_games}
+  end
```
Comment (Yetidancer) [inline lib/season_stats.rb]: wow

File: lib/game_collection.rb [hunk 4]
```diff
@@ -25,11 +22,27 @@ def game_lists_by_season
...
 
-  def games_by_season
+  def from_team(hash_of_games, team_id)
+    hash_of_games.reduce({}) do |acc, season_and_games|
+      acc[season_and_games[0]] = season_and_games[1].find_all {|game| game.home_team_id == team_id || game.away_team_id == team_id}
+      acc
+    end
+  end
+
+  def games_by_season #COUNT of games
```
Comment (Yetidancer) [inline lib/game_collection.rb]: helpful distinction

File: lib/season_stats.rb [hunk 5]
```diff
@@ -42,25 +42,20 @@ def make_season_game_array(season)
...
-      total_goals_scored: reg_games, total_goals_against: reg_games,
-      average_goals_scored: reg_games, average_goals_against: reg_games}
-    split_hash[:postseason] = {win_percentage: post_games,
-      total_goals_scored: post_games, total_goals_against: post_games,
-      average_goals_scored: post_games, average_goals_against: post_games}
+  def format_seasonal_summary(games)
+    games = "test"
+    {win_percentage: games,
+      total_goals_scored: games, total_goals_against: games,
+      average_goals_scored: games, average_goals_against: games}
```
Comment (Yetidancer) [inline lib/season_stats.rb]: looks much better

**Pull request merged**